runtime/drivers/duckdb/catalog.go

```diff
@@ -89,39 +97,37 @@ func (c *connection) CreateObject(ctx context.Context, instanceID string, obj *d
 		return err
 	}
 
-	obj.CreatedOn = now
-	obj.UpdatedOn = now
-	obj.RefreshedOn = now
+	e.CreatedOn = now
+	e.UpdatedOn = now
+	e.RefreshedOn = now
 	return nil
 }
 
-func (c *connection) UpdateObject(ctx context.Context, instanceID string, obj *drivers.CatalogObject) error {
-	// Serialize schema (note: if schema is nil, proto.Marshal returns nil)
-	schema, err := proto.Marshal(obj.Schema)
+func (c *connection) UpdateEntry(ctx context.Context, instanceID string, e *drivers.CatalogEntry) error {
+	// Serialize object
+	obj, err := proto.Marshal(e.Object)
 	if err != nil {
 		return err
 	}
 
 	_, err = c.db.ExecContext(
 		ctx,
-		"UPDATE rill.catalog SET type = ?, sql = ?, schema = ?, managed = ?, definition = ?, path = ?, refreshed_on = ?, updated_on = ? WHERE name = ?",
-		obj.Type,
-		obj.SQL,
-		schema,
-		obj.Managed,
-		obj.Definition,
-		obj.Path,
-		obj.RefreshedOn,
-		obj.UpdatedOn,
-		obj.Name,
+		"UPDATE rill.catalog SET type = ?, object = ?, path = ?, updated_on = ?, refreshed_on = ? WHERE name = ?",
+		e.Type,
+		obj,
+		e.Path,
+		e.UpdatedOn, // TODO: Use time.Now()
```

Comment:
The explicit use of `UpdatedOn` was intentional to make it the same as artifact's last updated

Reply:
Yeah, that's why I didn't change it here. It's still a little implicit, so keeping the TODO so we can fix at another time